Entities/ADoor.cpp

```diff
@@ -1,921 +1,598 @@
+
+	void ADoor::OpenDoor() {
+		if (m_DoorState == STOPPED) {
+			SharedDoorControls();
+			m_DoorState = CLOSING;
+			m_ChangedDirectionAfterStop = (m_DoorState == m_DoorStateOnStop) ? false : true;
+		}
+		if (m_DoorState == CLOSED || m_DoorState == CLOSING) {
+			SharedDoorControls();
+			m_DoorState = OPENING;
+		}
+		m_ResetDefaultTimer.Reset();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::CloseDoor() {
+		if (m_DoorState == STOPPED) {
+			SharedDoorControls();
+			m_DoorState = OPENING;
+			m_ChangedDirectionAfterStop = (m_DoorState == m_DoorStateOnStop) ? false : true;
+		}
+		if (m_DoorState == OPEN || m_DoorState == OPENING) {
+			SharedDoorControls();
+			m_DoorState = CLOSING;
+		}
+		m_ResetDefaultTimer.Reset();
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::StopDoor() {
+		if (m_DoorState == OPENING || m_DoorState == CLOSING) {
+			if (m_DrawMaterialLayerWhenOpen || m_DrawMaterialLayerWhenClosed) { DrawDoorMaterial(); }
+			m_DoorMoveStopTime = m_DoorMoveTime - m_DoorMoveTimer.GetElapsedSimTimeMS();
+			m_DoorStateOnStop = m_DoorState;
+			m_DoorMoveSound.Stop();
+			m_DoorMoveEndSound.Play(m_Pos);
+			m_DoorState = STOPPED;
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::SharedDoorControls() {
+		if (m_DoorState == OPEN || m_DoorState == CLOSED) {
+			m_DoorMoveStartSound.Play(m_Pos);
+			m_DoorMoveTimer.Reset();
+			if (m_DoorMaterialDrawn) { EraseDoorMaterial(); }
+			if (m_Door) { m_Door->DeepCheck(true); }
+		} else if (m_DoorState == OPENING || m_DoorState == CLOSING) {
+			m_DoorMoveSound.Stop();
+			if (!m_ResumeAfterStop) {
+				m_DoorDirectionChangeSound.Play(m_Pos);
+				m_DoorMoveTimer.SetElapsedSimTimeMS(m_DoorMoveTime - m_DoorMoveTimer.GetElapsedSimTimeMS());
+			} else {
+				m_DoorMoveStartSound.Play(m_Pos);
+				m_DoorMoveTimer.SetElapsedSimTimeMS(m_ChangedDirectionAfterStop ? m_DoorMoveTime - m_DoorMoveStopTime : m_DoorMoveStopTime);
+				m_ChangedDirectionAfterStop = false;
+				m_ResumeAfterStop = false;
+			}
+		} else if (m_DoorState == STOPPED) {
+			if (m_DoorMaterialDrawn) { EraseDoorMaterial(); }
+			m_ResumeAfterStop = true;
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ADoor::Update() {
+		Actor::Update();
+
+		if (m_Door && m_Door->IsAttached()) {
+			if (m_DoorState != STOPPED && m_SensorTimer.IsPastSimMS(m_SensorInterval)) { UpdateSensors(); }
+			UpdateDoorAttachable();
+		}
+
+		if (m_Door && !ApplyAttachableForces(m_Door)) {
+			// Remove the material representation if the door fell off
```

Comment:
Junk comment, the one explaining the logic on spinning isn't unreasonable, though I'd shift it and the code it refers to so that `m_Door = 0;` is above it.